examples/applications/plot_cyclical_feature_engineering.py

```diff
@@ -0,0 +1,741 @@
+
+ts_cv = TimeSeriesSplit(
+    n_splits=5, gap=48, max_train_size=10000, test_size=1000,
+)
+
+# %%
+# Let us manually inspect the various splits to check that the
+# `TimeSeriesSplit` works as we expect, starting with the first split:
+all_splits = list(ts_cv.split(X, y))
+train_0, test_0 = all_splits[0]
+
+# %%
+X.iloc[test_0]
+
+# %%
+X.iloc[train_0]
+
+# %%
+# We now inspect the last split:
+train_4, test_4 = all_splits[4]
+
+# %%
+X.iloc[test_4]
+
+# %%
+X.iloc[train_4]
+
+# %%
+# All is well. We are now ready to do some predictive modeling!
+#
+# Gradient Boosting
+# -----------------
+#
+# Gradient Boosting Regression with decision trees is often flexible enough to
+# efficiently handle heteorogenous tabular data with a mix of categorical and
+# numerical features as long as the number of samples is large enough.
+#
+# Here, we do minimal ordinal encoding for the categorical variables and then
+# let the model know that it should treat those as categorical variables by
+# using a dedicated tree splitting rule.
+#
+# The numerical variable need no preprocessing and, for the sake of simplicity,
+# we only try the default hyper-parameters for this model:
+from sklearn.pipeline import make_pipeline
+from sklearn.preprocessing import OrdinalEncoder
+from sklearn.compose import ColumnTransformer
+from sklearn.ensemble import HistGradientBoostingRegressor
+from sklearn.model_selection import cross_validate
+
+
+categorical_columns = [
+    "weather",
+    "season",
+    "holiday",
+    "workingday",
+]
+categories = [
+    ["clear", "misty", "rain", "heavy_rain"],
+    ["spring", "summer", "fall", "winter"],
+    ["False", "True"],
+    ["False", "True"],
+]
+ordinal_encoder = OrdinalEncoder(categories=categories)
+
+
+gbrt_pipeline = make_pipeline(
+    ColumnTransformer(
+        transformers=[
+            ("categorical", ordinal_encoder, categorical_columns),
+        ],
+        remainder="passthrough",
+    ),
+    HistGradientBoostingRegressor(
+        categorical_features=range(4),
+    ),
+)
+
+# %%
+#
+# Lets evaluate our gradient boosting model with the mean absolute error of the
+# relative demand averaged accross our 5 time-based cross-validation splits:
+
+
+def evaluate(model, X, y, cv):
+    cv_results = cross_validate(
+        model,
+        X,
+        y,
+        cv=ts_cv,
+        scoring=["neg_mean_absolute_error", "neg_root_mean_squared_error"]
+    )
+    mae = -cv_results["test_neg_mean_absolute_error"]
+    rmse = -cv_results["test_neg_root_mean_squared_error"]
+    print(
+        f"Mean Absolute Error:     {mae.mean():.3f} +/- {mae.std():.3f}\n"
+        f"Root Mean Squared Error: {rmse.mean():.3f} +/- {rmse.std():.3f}"
+    )
+
+
+evaluate(gbrt_pipeline, X, y, cv=ts_cv)
+
+# %%
+# This model has an average error around 4 to 5% of the maximum demand. This is
+# quite good for a first trial without any hyper-parameter tuning! We just had
+# to make the categorical variables explicit. Note that the time related
```

Comment:
I see that you mentioned this point now. I was expecting to see it a bit earlier :)

Reply:
I find more lightweight to do it this way.